File: src/main/java/com/example/progettoenterprise/dto/PrenotazioneDTO.java
package com.example.progettoenterprise.dto;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import jakarta.validation.constraints.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PrenotazioneDTO {
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Long id;

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private LocalDateTime dataPrenotazione;

    @NotNull(message = "Devi specificare il numero di persone")
    @Min(value=1, message = "Devi specificare un numero maggiore di 0")
    private Integer numeroPersone;

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Long viaggiatoreId;
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private String viaggiatoreUsername;

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Long viaggioId;
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private String viaggioTitolo;
}

File: frontendMobile/app/src/main/java/com/example/enterprisemobile/model/dto/PrenotazioneDTO.kt
package com.example.enterprisemobile.model

import com.google.gson.annotations.SerializedName

data class PrenotazioneDTO(
    @SerializedName("id") val id: Long,
    @SerializedName("dataPrenotazione") val dataPrenotazione: String?,
    @SerializedName("numeroPersone") val numeroPersone: Int,
    @SerializedName("viaggiatoreId") val viaggiatoreId: Long,
    @SerializedName("viaggiatoreUsername") val viaggiatoreUsername: String?,
    @SerializedName("viaggioId") val viaggioId: Long,
    @SerializedName("viaggioTitolo") val viaggioTitolo: String,
    @SerializedName("viaggioDataInizio") val viaggioDataInizio: String?,
    @SerializedName("viaggioDataFine") val viaggioDataFine: String?,
    @SerializedName("viaggioDestinazione") val viaggioDestinazione: String?,
    @SerializedName("stato") val stato: String // "IN_ATTESA", "CONFERMATA", "ANNULLATA"
)